examples/request.rs

```diff
@@ -1,50 +1,34 @@
-extern crate futures;
-extern crate pretty_env_logger;
-extern crate tmq;
-extern crate tokio;
-
-#[macro_use]
-extern crate log;
-
-extern crate failure;
-
-use futures::{stream, Future, Stream};
-
-use failure::Error;
-
-use tmq::*;
-
+use log::info;
 use std::env;
+use tmq::{request, Context, Multipart, Result};
 
-fn main() {
-    if let Err(_) = env::var("RUST_LOG") {
+#[tokio::main]
+async fn main() -> Result<()> {
+    if env::var("RUST_LOG").is_err() {
         env::set_var("RUST_LOG", "request=DEBUG");
     }
 
     pretty_env_logger::init();
 
-    let request = request(&Context::new())
-        .connect("tcp://127.0.0.1:7899")
-        .expect("Couldn't connect")
-        .with(make_request(5))
-        .for_each(|val| {
-            info!("Response: {}", val.as_str().unwrap_or(""));
-            Ok(())
-        })
-        .map_err(|err| {
-            error!("Error with request: {}", err);
-        });
-
-    tokio::run(request);
-}
-
-//Send some requests to the server
-fn make_request(count: usize) -> impl Stream<Item = Message, Error = Error> {
-    let mut vec = Vec::new();
-
-    for i in 0..count {
-        vec.push(Message::from(&format!("Request #{}", i)));
+    let mut send_sock = request(&Context::new())
+        .connect("tcp://127.0.0.1:7897")?
+        .finish()?;
+
+    let mut i = 0u32;
+    loop {
+        let message = format!("Req#{}", i);
+        i += 1;
+
+        info!("Request: {:?}", &message);
+        let mut multipart = Multipart::from(vec![zmq::Message::from(message.as_bytes())]);
+        let recv_sock = send_sock.send(&mut multipart).await?;
```

Comment:
@Kobzol ergonomics for a split sender-receiver object cluster seem pretty good to me.  Thoughts?  Also see `examples/reply.rs` for the opposite side of the loop.

Reply:
I like the interface! `examples/reply.rs` seems to be missing though?